packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmAnyTests.kt

```diff
@@ -16,39 +16,78 @@
 
 package io.realm.kotlin.test.shared
 
+import io.realm.kotlin.MutableRealm
 import io.realm.kotlin.Realm
 import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.Sample
+import io.realm.kotlin.entities.embedded.EmbeddedChild
+import io.realm.kotlin.entities.embedded.EmbeddedParent
+import io.realm.kotlin.entities.embedded.embeddedSchema
+import io.realm.kotlin.ext.asRealmObject
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.notifications.DeletedObject
+import io.realm.kotlin.notifications.SingleQueryChange
+import io.realm.kotlin.notifications.UpdatedObject
+import io.realm.kotlin.query.find
+import io.realm.kotlin.test.assertFailsWithMessage
 import io.realm.kotlin.test.platform.PlatformUtils
-import io.realm.kotlin.types.EmbeddedRealmObject
-import io.realm.kotlin.types.MutableRealmInt
-import io.realm.kotlin.types.ObjectId
+import io.realm.kotlin.test.util.use
+import io.realm.kotlin.types.RealmAny
 import io.realm.kotlin.types.RealmInstant
 import io.realm.kotlin.types.RealmObject
 import io.realm.kotlin.types.RealmUUID
+import io.realm.kotlin.types.annotations.Index
+import kotlinx.coroutines.async
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
 import org.mongodb.kbson.BsonObjectId
+import kotlin.reflect.KClass
 import kotlin.test.AfterTest
 import kotlin.test.BeforeTest
 import kotlin.test.Test
 import kotlin.test.assertContentEquals
 import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.time.Duration.Companion.seconds
 
 @Suppress("LargeClass")
 class RealmAnyTests {
 
+    private lateinit var configBuilder: RealmConfiguration.Builder
+    private lateinit var configuration: RealmConfiguration
     private lateinit var tmpDir: String
     private lateinit var realm: Realm
 
+    private val supportedTypesAndValues = listOf(
```

Comment:
Maybe add a `TODO` that we should somehow fail if this is not covering all our supported types ... Maybe we should tag it with a specific `TODO TYPEMETADATA` to be able to track the issues around the `TypeDescriptor` not being expressive enough.